cmd/spiffe-helper/main.go

```diff
@@ -13,29 +15,45 @@ func main() {
 	// 1. Create Sidecar
 	// 2. Run Sidecar's Daemon
 
+	log := logger.Std
 	configFile := flag.String("config", "helper.conf", "<configFile> Configuration file path")
 	flag.Parse()
 
 	config, err := ParseConfig(*configFile)
 	if err != nil {
-		log.Fatalf("error parsing configuration file: %v\n%v", *configFile, err)
+		log.Errorf("error parsing configuration file: %v\n%v", *configFile, err)
+		panic(err)
+	}
+	timeout, err := GetTimeout(config)
+	if err != nil {
+		log.Errorf("error parsing timeout: %s\n%v", config.Timeout, err)
+		panic(err)
 	}
+	config.Log = log
 
-	log.Printf("Connecting to agent at %q\n", config.AgentAddress)
+	log.Infof("Connecting to agent at %q\n", config.AgentAddress)
 	if config.Cmd == "" {
-		log.Println("Warning: no cmd defined to execute.")
+		log.Warnf("No cmd defined to execute.")
 	}
 
-	log.Printf("Using configuration file: %q\n", *configFile)
+	log.Infof("Using configuration file: %q\n", *configFile)
 
-	spiffeSidecar, err := sidecar.NewSidecar(config)
+	spiffeSidecar := sidecar.NewSidecar(config)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	err = spiffeSidecar.RunDaemon(ctx)
 	if err != nil {
 		panic(err)
 	}
 
-	ctx := context.Background()
-	err = spiffeSidecar.RunDaemon(ctx)
-	if err != nil {
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt)
```

Comment:
We should instead use signal.NotifyContext to cancel the context when the signal is hit.